src/script/descriptor.cpp

```diff
@@ -94,9 +94,9 @@ class ConstPubkeyProvider final : public PubkeyProvider
 
 public:
     ConstPubkeyProvider(const CPubKey& pubkey) : m_pubkey(pubkey) {}
-    bool GetPubKey(int pos, const SigningProvider& arg, CPubKey& key, KeyOriginInfo& info) const override
+    bool GetPubKey(int pos, const SigningProvider& arg, CPubKey* key, KeyOriginInfo& info) const override
     {
-        key = m_pubkey;
+        if (key) *key = m_pubkey;
```

Comment:
How about instead extracting `GetPubKeyInfo` and changing up the caller in `ExpandHelper` to only request the desired information? This looks doable if `cache_read` and `cache_write` are mutually exclusive.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14646#discussion_r235786983

> How about instead extracting GetPubKeyInfo and changing up the caller in ExpandHelper to only request the desired information? This looks doable if cache_read and cache_write are mutually exclusive.

It does seem like it would be a minor simplification to have separate methods for retrieving CPubKey and KeyOriginInfo, since KeyOriginInfo isn't needed by ExpandHelper. But that would make this PR bigger, and I also don't see what it has to do with cache read/write becoming exclusive options.